Negative numbers issue

I tried to compile the following program:

```
package main

import "external"

func main() {
	foo := -4
	external.Printf("foo is = %d\n", foo)
}
```
It failed with the following panic

```
2019/03/23 19:40:45 unable to handle default: 16 - {Type:OPERATOR, Val:-, Line:6}
panic: unable to handle default: 16 - {Type:OPERATOR, Val:-, Line:6}
```

--- a/compiler/compiler/compiler.go
+++ b/compiler/compiler/compiler.go
@@ -268,6 +268,8 @@ func (c *Compiler) compileValue(node parser.Node) value.Value {
 		return c.compileConstantNode(v)
 	case *parser.OperatorNode:
 		return c.compileOperatorNode(v)
+	case *parser.SubNode:
+		return c.compileSubNode(v)
 	case *parser.NameNode:
 		return c.varByName(v.Name)
 	case *parser.CallNode:

--- a/compiler/compiler/condition.go
+++ b/compiler/compiler/condition.go
@@ -112,6 +112,26 @@ func (c *Compiler) compileOperatorNode(v *parser.OperatorNode) value.Value {
 	}
 }
 
+func (c *Compiler) compileSubNode(v *parser.SubNode) value.Value {
+	right := c.compileValue(v.Item)
+	rVal := right.Value
+
+	if right.IsVariable {
+		rVal = c.contextBlock.NewLoad(rVal)
+	}
+
+	res := c.contextBlock.NewSub(
+		constant.NewInt(rVal.Type().(*llvmTypes.IntType), 0),
+		rVal,
+	)
+
+	return value.Value {
+		Value: res,
+		Type: right.Type,
+		IsVariable: false,
+	}
+}
+
 func (c *Compiler) compileConditionNode(v *parser.ConditionNode) {
 
 	c.pushVariablesStack()

--- a/compiler/parser/node.go
+++ b/compiler/parser/node.go
@@ -361,6 +361,15 @@ func (nn NegateNode) String() string {
 	return fmt.Sprintf("NegateNode-!%s", nn.Item)
 }
 
+type SubNode struct {
+	baseNode
+	Item Node
+}
+
+func (sn SubNode) String() string {
+	return fmt.Sprintf("-%s", sn.Item)
+}
+
 type InitializeSliceNode struct {
 	baseNode
 	Type  TypeNode

--- a/compiler/parser/parser.go
+++ b/compiler/parser/parser.go
@@ -105,6 +105,12 @@ func (p *parser) parseOne(withAheadParse bool) (res Node) {
 			return
 		}
 
+		if current.Val == "-" {
+			p.i++
+			res = p.aheadParse(&SubNode{Item: p.parseOne(true)})
+			return
+		}
+
 		// Slice or array initalization
 		if current.Val == "[" {
 			next := p.lookAhead(1)

--- a/compiler/testdata/negative-number.go
+++ b/compiler/testdata/negative-number.go
@@ -1,0 +1,28 @@
+package main
+
+import "external"
+
+
+func baz() int32 {
+	return int32(-18)
+}
+
+
+func main() {
+	foo := -4
+	// foo is = -4
+	// -foo is = 4
+	external.Printf("foo is = %d\n", foo)
+	external.Printf("-foo is = %d\n", -foo)
+
+	// bar is = -16
+	a := 16
+	bar := -a
+	external.Printf("bar is = %d\n", bar)
+
+	// baz is = -18
+	external.Printf("baz is = %d\n", baz())
+
+	// -baz is = 18
+	external.Printf("-baz is = %d\n", -baz())
+}

